google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerException.java

```diff
@@ -197,4 +197,9 @@ public ErrorDetails getErrorDetails() {
     }
     return null;
   }
+
+  /** Sets the requestId. This method is meant to be used internally and not by customers. */
+  public void setRequestId(XGoogSpannerRequestId reqId) {
```

Comment:
nit: (for a follow-up PR) You can additionally use the `@InternalApi` annotation to indicate this.